src/redux/slices/RegisterSlice.ts

```diff
@@ -0,0 +1,68 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+import { URL } from '@/utils';
+import { RegisterError } from '@/app/(Authentication)/register/page';
+
+export interface UserInterface {
+  name: string;
+  email: string;
+  password: string;
+  phone: string;
+  address: string;
+}
+
+interface RegistrationState {
+  loading: boolean;
+  error: RegisterError | null;
+  success: boolean;
+}
+
+const initialState: RegistrationState = {
+  loading: false,
+  error: null,
+  success: false
+};
+
+const API_URL = `${URL}/api/users/register`;
+
+export const registerUser = createAsyncThunk(
+  'authentication/registerUser',
+  async (userData: UserInterface, { rejectWithValue }) => {
+    try {
```

Comment:
configure axios instance make it easy to use it only one thing must be passed on header, endpoint, and other args.

Reply:
Done 👍